src/v2/Artsy/Relay/middleware/cacheMiddleware.ts

```diff
@@ -0,0 +1,152 @@
+import { QueryResponseCache } from "relay-runtime"
+import { isFunction } from "lodash"
+import createLogger from "v2/Utils/logger"
+
+const logger = createLogger("v2/Artsy/middleware/cacheMiddleware")
+
+interface CacheMiddlewareOpts {
+  size?: number
+  ttl?: number
+  onInit?: (cache: QueryResponseCache) => any
+  allowMutations?: boolean
+  allowFormData?: boolean
+  clearOnMutation?: boolean
+  cacheErrors?: boolean
+}
+
+class Cache {
+  relayCache: any // FIXME: add type
+  redisCache: any // FIXME: add type
+  isServer: boolean
+
+  constructor(opts) {
+    this.relayCache = new QueryResponseCache(opts)
+    this.isServer = typeof window === "undefined"
+
+    if (this.isServer) {
+      const { promisify } = require("util")
+      const redis = require("redis")
+      const client = redis.createClient()
```

Comment:
For illustration I wanted to keep this bare bones, but note we already have an (old) redis [cache util](https://github.com/artsy/force/blob/5c59333da35ad9b208c8da43dc8501cc158d819c/src/lib/cache.coffee#L17).